TombEngine/Game/effects/effects.cpp

```diff
@@ -201,69 +208,87 @@ void UpdateSparks()
 
 	for (int i = 0; i < MAX_PARTICLES; i++)
 	{
-		auto* spark = &Particles[i];
+		auto& spark = Particles[i];
 
-		if (spark->on)
+		if (spark.on)
 		{
-			spark->StoreInterpolationData();
+			spark.StoreInterpolationData();
 
-			spark->life--;
+			spark.life--;
 
-			if (!spark->life)
+			if (!spark.life)
 			{
-				if (spark->dynamic != -1)
-					ParticleDynamics[spark->dynamic].On = false;
+				if (spark.dynamic != -1)
+					ParticleDynamics[spark.dynamic].On = false;
 
-				spark->on = false;
+				spark.on = false;
+				continue;
+			}
+
+			if (spark.SpriteSeqID == ID_WATERFALL &&
+				spark.y >= spark.targetPos.y && 
+				spark.targetPos.w == spark.fxObj)
```

Comment:
Why item number is packed into `w` component of `targetPos`, and why it is compared here with `fxObj` field of the very same spark? Can spark ownership change in runtime, or it's just duplicated fields?